lib/src/sql/constant.rs

```diff
@@ -106,51 +105,27 @@ impl Serialize for Constant {
-			}
-		} else {
-			match self {
-				Self::MathE => s.serialize_f64(std::f64::consts::E),
-				Self::MathFrac1Pi => s.serialize_f64(std::f64::consts::FRAC_1_PI),
-				Self::MathFrac1Sqrt2 => s.serialize_f64(std::f64::consts::FRAC_1_SQRT_2),
-				Self::MathFrac2Pi => s.serialize_f64(std::f64::consts::FRAC_2_PI),
-				Self::MathFrac2SqrtPi => s.serialize_f64(std::f64::consts::FRAC_2_SQRT_PI),
-				Self::MathFracPi2 => s.serialize_f64(std::f64::consts::FRAC_PI_2),
-				Self::MathFracPi3 => s.serialize_f64(std::f64::consts::FRAC_PI_3),
-				Self::MathFracPi4 => s.serialize_f64(std::f64::consts::FRAC_PI_4),
-				Self::MathFracPi6 => s.serialize_f64(std::f64::consts::FRAC_PI_6),
-				Self::MathFracPi8 => s.serialize_f64(std::f64::consts::FRAC_PI_8),
-				Self::MathLn10 => s.serialize_f64(std::f64::consts::LN_10),
-				Self::MathLn2 => s.serialize_f64(std::f64::consts::LN_2),
-				Self::MathLog102 => s.serialize_f64(std::f64::consts::LOG10_2),
-				Self::MathLog10E => s.serialize_f64(std::f64::consts::LOG10_E),
-				Self::MathLog210 => s.serialize_f64(std::f64::consts::LOG2_10),
-				Self::MathLog2E => s.serialize_f64(std::f64::consts::LOG2_E),
-				Self::MathPi => s.serialize_f64(std::f64::consts::PI),
-				Self::MathSqrt2 => s.serialize_f64(std::f64::consts::SQRT_2),
-				Self::MathTau => s.serialize_f64(std::f64::consts::TAU),
-			}
-		}
+        match self {
+            Self::MathE => s.serialize_unit_variant(TOKEN, 0, "MathE"),
+            Self::MathFrac1Pi => s.serialize_unit_variant(TOKEN, 1, "MathFrac1Pi"),
+            Self::MathFrac1Sqrt2 => s.serialize_unit_variant(TOKEN, 2, "MathFrac1Sqrt2"),
+            Self::MathFrac2Pi => s.serialize_unit_variant(TOKEN, 3, "MathFrac2Pi"),
+            Self::MathFrac2SqrtPi => s.serialize_unit_variant(TOKEN, 4, "MathFrac2SqrtPi"),
+            Self::MathFracPi2 => s.serialize_unit_variant(TOKEN, 5, "MathFracPi2"),
+            Self::MathFracPi3 => s.serialize_unit_variant(TOKEN, 6, "MathFracPi3"),
+            Self::MathFracPi4 => s.serialize_unit_variant(TOKEN, 7, "MathFracPi4"),
+            Self::MathFracPi6 => s.serialize_unit_variant(TOKEN, 8, "MathFracPi6"),
+            Self::MathFracPi8 => s.serialize_unit_variant(TOKEN, 9, "MathFracPi8"),
+            Self::MathLn10 => s.serialize_unit_variant(TOKEN, 10, "MathLn10"),
+            Self::MathLn2 => s.serialize_unit_variant(TOKEN, 11, "MathLn2"),
+            Self::MathLog102 => s.serialize_unit_variant(TOKEN, 12, "MathLog102"),
+            Self::MathLog10E => s.serialize_unit_variant(TOKEN, 13, "MathLog10E"),
+            Self::MathLog210 => s.serialize_unit_variant(TOKEN, 14, "MathLog210"),
+            Self::MathLog2E => s.serialize_unit_variant(TOKEN, 15, "MathLog2E"),
+            Self::MathPi => s.serialize_unit_variant(TOKEN, 16, "MathPi"),
+            Self::MathSqrt2 => s.serialize_unit_variant(TOKEN, 17, "MathSqrt2"),
+            Self::MathTau => s.serialize_unit_variant(TOKEN, 18, "MathTau"),
+        }
```

Comment:
Is there some reason this can't be automatically derived? At first glance, it looks like what `derive(serde::Serialize)` would generate, or does `#[serde(rename = "token...")]` not work on `enum`?

Reply:
@finnbear @rushmorem said:

> I left enums manually implemented for now as that is the safest bet when we start using variant indices. We can easily derive the enums as well but once we start using variant indices to minimise storage size, we would need to have a policy of only adding new enum variants at the bottom, otherwise the indices wouldn't be stable.

So for enums, we will either leave it as is, or define a policy where new enums are only ever added at the end of the enum variants. We'll do this in a separate PR.

Reply:
If we derive, we should put a comment at the bottom of all enums (at the bottom of all variants)...

```rs
// Add new variants here
```

Reply:
Alright; as there is a good reason, I don't have an objection.